src/diffusers/configuration_utils.py

```diff
@@ -526,7 +526,8 @@ def extract_init_dict(cls, config_dict, **kwargs):
                 init_dict[key] = config_dict.pop(key)
 
         # 4. Give nice warning if unexpected values have been passed
-        if len(config_dict) > 0:
+        only_quant_config_remaining = len(config_dict) == 1 and "quantization_config" in config_dict
```

Comment:
Because `quantization_config` isn't a part of any model's `__init__()`.

Reply:
We cannot remove `quantization_config` from the config of a model as that would prevent loading of the quantized models via `from_pretrained()`. 

`quantization_config` isn't used for initializing a model, it's used to determine what kind of quantization configuration to inject inside the given model. This is why it's only used in `from_pretrained()` of `ModelMixin`. 

LMK if you have a better idea to handle it.

Reply:
we do not remove them from the config, just not add to the `config_dict` inside this `extract_init_dict` method: basically, the `cofig_dict` in this function goes through these steps: 
1. it is used to create `init_dict`: the quantisation config will not go there it is not affected if we do not add it to `config_dict`
2. it is used to throw a warning after we created`init_dict`, if the quantisation configs were not there, we do not need to throw a warning for it
3. it goes into `unused_kwargs` - so I think this is the only difference it would make, do we need the quantisation config to be in `unused_kwargs` returned by `extract_init_dict`? I think `unused_kwargs` is only used to send warnings for unexpected stuff, but since quantisation config is expected, but just not needed for initialization, I think it does not need to go to the `unused_kwargs`?

```diff
    @classmethod
    def extract_init_dict(cls, config_dict, **kwargs):
         ...
        config_dict = {k: v for k, v in config_dict.items() if k not in used_defaults and k != "_use_default_values"

        # remove private attributes
        config_dict = {k: v for k, v in config_dict.items() if not k.startswith("_")}
        
+      # remove quantization_config
+      config_dict = {k: v for k, v in config_dict.items() if k != "quantization_config")}
        
        
        ## here we use config_dict to create `init_dict` which will be passed to `__init__` method
        init_dict = {}
        for key in expected_keys:
                 ...
                init_dict[key] = config_dict.pop(key)
-      only_quant_config_remaining = len(config_dict) == 1 and "quantization_config" in config_dict
-      if len(config_dict) > 0 and not only_quant_config_remaining:
+     if len(config_dict) > 0:
            logger.warning(
                f"The config attributes {config_dict} were passed to {cls.__name__}, "
                "but are not expected and will be ignored. Please verify your "
                f"{cls.config_name} configuration file."
            )
       ....
        # 6. Define unused keyword arguments
        unused_kwargs = {**config_dict, **kwargs}

        return init_dict, unused_kwargs, hidden_config_dict
```